src/node/server/index.ts

```diff
@@ -76,7 +76,8 @@ export function createServer(config: ServerConfig): Server {
     moduleRewritePlugin,
     htmlRewritePlugin,
     // user plugins
-    ...(Array.isArray(configureServer) ? configureServer : [configureServer]),
+    // configureServer already normalize by ./config.ts
+    ...(configureServer as ServerPlugin[]),
```

Comment:
When using the `createServer` directly as an API, it's possible to pass in `configureServer` as a single function instead of an Array.